crates/configuration/src/version1.rs

```diff
@@ -135,8 +158,21 @@ async fn select_first_row(
 }
 
 /// Construct the deployment configuration by introspecting the database.
-pub async fn configure(configuration: &RawConfiguration) -> Result<RawConfiguration, Error> {
-    let mssql_pool = create_mssql_pool(configuration).await.unwrap();
+pub async fn configure(
+    file_path: &std::path::Path,
+    configuration: &RawConfiguration,
+    environment: impl Environment,
+) -> Result<RawConfiguration, Error> {
+    let connection_string = match &configuration.mssql_connection_string {
+        uri::ConnectionUri(Secret::Plain(s)) => s.to_owned(),
+        uri::ConnectionUri(Secret::FromEnvironment { variable }) => environment
+            .read(variable)
+            .map_err(|err| Error::MissingEnvironmentVariable {
+                file_path: file_path.to_path_buf(),
+                message: format!("{err}"),
+            })?,
+    };
+    let mssql_pool = create_mssql_pool(&connection_string).await.unwrap();
```

Comment:
Can we avoid the use of `unwrap` here and instead throw an error?

Reply:
Updated